File: latihan1/Java/Main.java
//import library
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    //membuat scanner
    private static Scanner input = new Scanner(System.in);

    //method untuk meminta input data mahasiswa dari user
    private static Mahasiswa inputMahasiswa() {
        System.out.print("Masukkan nama: ");
        String nama = input.nextLine();
        System.out.print("Masukkan NIM: ");
        String NIM = input.nextLine();
        System.out.print("Masukkan program studi: ");
        String programStudi = input.nextLine();
        System.out.print("Masukkan fakultas: ");
        String fakultas = input.nextLine();
        return new Mahasiswa(nama, NIM, programStudi, fakultas);
    }

    //method untuk menampilkan menu yang dapat dipilih user
    public static void main(String[] args) {
        DaftarMahasiswa daftar = new DaftarMahasiswa();
        int pilihan = 0;
        do {
            System.out.println("\nMenu:");
            System.out.println("1. Tambah mahasiswa");
            System.out.println("2. Ubah data mahasiswa");
            System.out.println("3. Hapus data mahasiswa");
            System.out.println("4. Tampilkan daftar mahasiswa");
            System.out.println("0. Keluar");
            System.out.print("Masukkan pilihan: ");
            pilihan = input.nextInt();
            input.nextLine();
            switch (pilihan) {
                //menambahkan daftar mahasiswa
                case 1:
                    daftar.tambahMahasiswa(inputMahasiswa());
                    break;
                //mengubah data mahasiswa
                case 2:
                    daftar.tampilkanDaftar();
                    System.out.print("Masukkan nomor urut mahasiswa yang akan diubah: ");
                    int index = input.nextInt();
                    input.nextLine();
                    daftar.ubahMahasiswa(index - 1, inputMahasiswa());
                    break;
                //menghapus data mahasiswa
                case 3:
                    daftar.tampilkanDaftar();
                    System.out.print("Masukkan nomor urut mahasiswa yang akan dihapus: ");
                    index = input.nextInt();
                    input.nextLine();
                    daftar.hapusMahasiswa(index - 1);
                    break;
                //menampilkan daftar mahasiswa
                case 4:
                    daftar.tampilkanDaftar();
                    break;
                //keluar dari program
                case 0:
                    break;
                //pilihan angka selain yang ada di menu
                default:
                    System.out.println("Pilihan tidak valid! Mohon pilih nomor 1 - 4.");
                    break;
            }
        } while (pilihan != 0);
        System.out.println("Terima kasih telah menggunakan program ini.");
    }
}

File: latihan1/main.py
#class mahasiswa untuk objek mahasiswa
class Mahasiswa:
    def __init__(self, nama, nim, prodi, fakultas):
        self.nama = nama
        self.nim = nim
        self.prodi = prodi
        self.fakultas = fakultas

    def __str__(self):
        return f"Nama \t\t: {self.nama}\nNIM \t\t: {self.nim}\nProdi \t\t: {self.prodi}\nFakultas \t: {self.fakultas}\n"

#class DaftarMahasiswa untuk mengelola daftar mahasiswa
class DaftarMahasiswa:
    def __init__(self):
        self.daftar_mahasiswa = []

    #method untuk menambah daftar mahasiswa
    def tambah_mahasiswa(self, mahasiswa):
        self.daftar_mahasiswa.append(mahasiswa)

    #method untuk mengubah data mahasiswa
    def ubah_mahasiswa(self, nim, mahasiswa_baru):
        for i, mahasiswa in enumerate(self.daftar_mahasiswa):
            if mahasiswa.nim == nim:
                self.daftar_mahasiswa[i] = mahasiswa_baru

    #method untuk menghapus data mahasiswa
    def hapus_mahasiswa(self, nim):
        for i, mahasiswa in enumerate(self.daftar_mahasiswa):
            if mahasiswa.nim == nim:
                del self.daftar_mahasiswa[i]

    #method untuk menampikan daftar mahasiswa
    def tampilkan_daftar(self):
        if not self.daftar_mahasiswa:
            print("Tidak ada mahasiswa yang terdaftar.")
        else:
            print("Daftar Mahasiswa Universitas Pendidikan Indonesia\n")
            for i, mahasiswa in enumerate(self.daftar_mahasiswa):
                print(f"ID \t\t\t: {i+1}")
                print(mahasiswa)

#fungsi untuk meminta input data mahasiswa dari user
def input_mahasiswa():
    nama = input("Masukkan nama: ")
    nim = input("Masukkan NIM: ")
    prodi = input("Masukkan program studi: ")
    fakultas = input("Masukkan fakultas: ")
    return Mahasiswa(nama, nim, prodi, fakultas)

if __name__ == "__main__":
    #membuat objek DaftarMahasiswa
    daftar_mahasiswa = DaftarMahasiswa()
    while True:
        #menampilkan menu program yang dapat dipilih user
        print("\nMenu:")
        print("1. Tambah Mahasiswa")
        print("2. Ubah Data Mahasiswa")
        print("3. Hapus Mahasiswa")
        print("4. Tampilkan Daftar Mahasiswa")
        print("5. Keluar")
        pilihan = int(input("Masukkan pilihan: "))

        if pilihan == 1:
            #menambahkan mahasiswa ke dalam daftar mahasiswa
            mahasiswa = input_mahasiswa()
            daftar_mahasiswa.tambah_mahasiswa(mahasiswa)
            print("Mahasiswa berhasil ditambahkan.")
        elif pilihan == 2:
            #mengubah data mahasiswa
            daftar_mahasiswa.tampilkan_daftar()
            nim = input("Masukkan nomor urut mahasiswa yang ingin diubah: ")
            mahasiswa_baru = input_mahasiswa()
            daftar_mahasiswa.ubah_mahasiswa(nim, mahasiswa_baru)
            print("Data mahasiswa berhasil diubah!")
        elif pilihan == 3:
            #menghapus data mahasiswa
            daftar_mahasiswa.tampilkan_daftar()
            nim = input("Masukkan nomor urut mahasiswa yang ingin dihapus: ")
            daftar_mahasiswa.hapus_mahasiswa(nim)
            print("Mahasiswa berhasil dihapus!")
        elif pilihan == 4:
            #menampilkan daftar mahasiswa
            daftar_mahasiswa.tampilkan_daftar()
        elif pilihan == 5:
            #keluar dari program
            break
        else:
            #ketika user memilih pilihan selain daripada yang ada di menu
            print("Pilihan tidak valid! Mohon pilih nomor 1 - 5")
